example-app/SantokuApp/src/screens/demo/push-notification/PushNotificationScreen.tsx

```diff
@@ -1,53 +1,81 @@
-import {CompositeScreenProps} from '@react-navigation/native';
-import {NativeStackScreenProps} from '@react-navigation/native-stack';
-import {DemoStackParamList, RootStackParamList} from 'navigation/types';
-import React from 'react';
+import {DemoStackParamList} from 'navigation/types';
+import React, {useEffect} from 'react';
 import {SafeAreaView, ScrollView, StyleSheet, View} from 'react-native';
 import {Button, Divider, Text} from 'react-native-elements';
 
+import {SelectPicker} from '../../../components/picker';
+import {log} from '../../../framework';
+import {useSpecAndSourceCodeLink} from '../useSpecAndSouceCodeLink';
 import {usePushNotification} from './usePushNotification';
 
-type Props = CompositeScreenProps<
-  NativeStackScreenProps<DemoStackParamList, typeof ScreenName>,
-  NativeStackScreenProps<RootStackParamList>
->;
-
 const ScreenName = 'PushNotification';
-const Screen: React.FC<Props> = ({navigation}) => {
-  const {authStatus, token, requestUserPermission, getToken, notifyMessageToAll, notifyMessageToMe, openSettings} =
-    usePushNotification();
+const Screen: React.FC = () => {
+  const {
+    authStatus,
+    token,
+    getPermissionStatus,
+    requestUserPermissionWithoutOptions,
+    requestUserPermissionWithProvisional,
+    getToken,
+    notifyMessageToAll,
+    notifyMessageToMe,
+    openSettings,
+    channels,
+    channelKey,
+    onSelectedChannelChange,
+  } = usePushNotification();
+
+  useEffect(() => {
+    getPermissionStatus().catch(e => log.trace(`Failed to get permission status. cause=[${String(e)}]`));
+    getToken().catch(e => log.trace(`Failed to get token. cause=[${String(e)}]`));
+  }, [getPermissionStatus, getToken]);
+
+  const {SpecAndSourceCodeLink} = useSpecAndSourceCodeLink('push-notification');
 
   return (
     <SafeAreaView style={styles.container}>
       <ScrollView contentInsetAdjustmentBehavior="automatic">
+        <SpecAndSourceCodeLink />
+        <Divider orientation="vertical" style={styles.divider} />
         <View>
-          <Text>【Permissionのステータス】</Text>
+          <Text>【現在のPermissionのステータス】</Text>
           <Text>{authStatus ?? 'Permissionのステータスが表示されます'}</Text>
         </View>
-        <View style={styles.buttonContainer}>
-          <Button onPress={requestUserPermission} title="Permissionの許可" />
+        <View style={styles.elementContainer}>
+          <Button onPress={requestUserPermissionWithoutOptions} title="Permissionの許可" />
+        </View>
+        <Text>一度でも許可ダイアログで許可、もしくは拒否をすると仮許可はできません</Text>
```

Comment:
Please keep the text wording aline with the documentation.

Reply:
a9cae386734f3bee34c692599a9eeded509e524f